### PR TITLE
global assertion object for orientation

### PR DESCRIPTION
**Pull request by ellenmobify**
Assertions object to check orientation.

JIRA: https://mobify.atlassian.net/browse/CSOG-41

## Changes
- Adds `global` assertions object for any context
- Assertions for device orientation
- Example test that uses the assertion for orientation

## How to test-drive this PR
- Clone https://github.com/mobify/generator-astro
- Change `generator.sh`:
```
SCAFFOLD_VERSION="CSOG-41__orientation"
SCAFFOLD_URL="https://github.com/mobify/astro-scaffold/archive/CSOG-41__orientation.zip"
```
- Run `./generator.sh`
- `cd` into generated project and run `./scripts/ios-appium.sh`

## TODOS:
- [x] Tests have been written.
- [x] Tests are passing on your machine.
- [x] Change works in both Android and iOS.
- [x] Scaffold is working. If a new feature was added, it was added to the Scaffold app.
- [x] Run the generator to make sure it still functions correctly.
- [x] +1 from an engineer on the Astro team.
- [x] +1 from QA @lilyzh

### CODE REVIEW

File: app/tests/system/assertions/global.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/*
+ Assert that landscape mode is disabled.
+*/
+GlobalAssertions.prototype.assertPortrait = function() {
+    var self = this;
+    this.browser
+        .contexts(function(result) {
+            self.browser
+                .setContext(result.value[result.value.length - 1])
+                .setOrientation('LANDSCAPE')
```
Comment (lilyzh) [inline app/tests/system/assertions/global.js]: Should it be ".setOrientation('PORTRAIT')"?

File: app/tests/system/assertions/global.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                .execute(function() {
+                    return window.orientation;
+                }, function(result) {
+                        var rotation = result.value;
+                        self.browser.log('Window rotation should be 0.');
```
Comment (lilyzh) [inline app/tests/system/assertions/global.js]: Can we put more info here, like what does 0 mean, Portrait or Landscape?

File: app/tests/system/assertions/global.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    this.browser
+        .contexts(function(result) {
+            self.browser
+                .setContext(result.value[result.value.length - 1])
+                .setOrientation('LANDSCAPE')
+                .execute(function() {
+                    return window.orientation;
+                }, function(result) {
+                        var rotation = result.value;
+                        self.browser.log('Window rotation should not be 0.');
```
Comment (lilyzh) [inline app/tests/system/assertions/global.js]: Can we put more info here, like what does 0 mean, Portrait or Landscape?

**Comment by jansepar** [comment]
Hey folks - FYI, reviewers for any upstream contributions to Astro and Astro-related repos should include members of the Astro team. cc @jeremywiebe

File: app/tests/system/assertions/global.js (hunk 1 above)
Comment (ellenmobify) [inline app/tests/system/assertions/global.js]: Nope. We set the orientation to LANDSCAPE and check that the value of `window.orientation` is unchanged.

**Comment by lilyzh** [comment]
Looks good. :+1:

File: app/tests/system/assertions/global.js (hunk 2 above)
Comment (jeremywiebe) [inline app/tests/system/assertions/global.js]: Maybe we should have a javascript "enum" at the top of the file?  
```
var Orientations: { 
   Portrait: 0,
   // etc.
};
```

**Comment by ellenmobify** [comment]
@jeremywiebe Mind taking another look?

**Comment by jeremywiebe** [comment]
Looks good!  :+1:

**Pull request merged**